lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -170,95 +177,122 @@ def install_libraries
+            end]
 
-            add_system_framework_dependencies
+            return [pod_target_installation_results, aggregate_target_installation_results]
           end
         end
 
-        def integrate_targets
-          pod_targets_to_integrate = pod_targets.select { |pt| !pt.test_native_targets.empty? || pt.contains_script_phases? }
+        def integrate_targets(pod_target_installation_results)
+          pod_targets_to_integrate = pod_target_installation_results.values.select do |pod_target_installation_result|
+            pod_target = pod_target_installation_result.target
+            !pod_target_installation_result.test_native_targets.empty? || pod_target.contains_script_phases?
+          end
           unless pod_targets_to_integrate.empty?
             UI.message '- Integrating targets' do
-              pod_targets_to_integrate.each do |pod_target|
-                PodTargetIntegrator.new(pod_target).integrate!
+              pod_targets_to_integrate.each do |pod_target_installation_result|
+                PodTargetIntegrator.new(pod_target_installation_result).integrate!
               end
             end
           end
         end
 
-        def add_system_framework_dependencies
-          # @TODO: Add Specs
-          pod_targets.select(&:should_build?).sort_by(&:name).each do |pod_target|
-            test_file_accessors, file_accessors = pod_target.file_accessors.partition { |fa| fa.spec.test_specification? }
-            file_accessors.each do |file_accessor|
-              add_system_frameworks_to_native_target(file_accessor, pod_target.native_target)
-            end
-            test_file_accessors.each do |test_file_accessor|
-              native_target = pod_target.native_target_for_spec(test_file_accessor.spec)
-              add_system_frameworks_to_native_target(test_file_accessor, native_target)
+        def add_system_framework_dependencies(pod_target_installation_results)
+          pod_target_installation_results.each do |target_installation_result|
+            pod_target = target_installation_result.target
+            next unless pod_target.should_build?
+            pod_target.file_accessors.each do |file_accessor|
+              native_target = target_installation_result.native_target_for_spec(file_accessor.spec)
+              add_system_frameworks_to_native_target(file_accessor, native_target)
             end
           end
         end
 
         # Adds a target dependency for each pod spec to each aggregate target and
         # links the pod targets among each other.
         #
+        # @param  [Array[Hash{String=>TargetInstallationResult}]] target_installation_results
+        #         the installation results array that was produced including pod target installation results and
+        #         aggregate target installation results.
+        #
         # @return [void]
         #
-        def set_target_dependencies
+        def wire_target_dependencies(target_installation_results)
```

Comment:
recommend side by side diff here, this method is verbose but it's in my opinion *way* simpler to understand, it also helps dramatically with performance for large projects.

Reply:
This method is mostly unreadable now :/ - There are a few cases within it where things can be broken out into their own methods without meaningful perf impact, that would improve readability and make testing easier.